### PR TITLE
Feature/generating page

### PR DESCRIPTION
**Pull request by tomaszgil**
Add gulp setup, templating engine, refactor styling for components (apart from footer and events)

### CODE REVIEW

File: src/scss/_about.scss [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
   }
-}
 
-@media all and (max-width: 1000px) {  // one column layout
-  [class='about'] {
-    .about-header,
-    .about-columns {
-      margin-left: 15%;
-      width: 70%;
+  @media all and (max-width: 768px) { // one column layout
```
Comment (rrudol) [inline src/scss/_about.scss]: It should be a mixin

File: src/scss/_contact.scss [hunk 2]
```diff
@@ -1,87 +1,78 @@
...
+  text-align: center;
   width: 100%;
 
-  .contact-box {
-    margin: 0 auto;
-    max-width: 65rem;
-    text-align: center;
-    width: 80%;
+  .contact-heading {
+    font-size: 2.6rem;
```
Comment (rrudol) [inline src/scss/_contact.scss]: use @include font-size(2.6);

File: src/scss/_contact.scss [hunk 3]
```diff
@@ -1,87 +1,78 @@
...
-        }
+      background: transparent;
+      border: 0;
+      border-bottom: $border;
+      box-sizing: border-box;
+      color: $text-color;
+      font-family: inherit;
+      opacity: 0.8;
+      padding: 12px;
+      transition: all 0.3s ease-out;
```
Comment (rrudol) [inline src/scss/_contact.scss]: Is transition for all properties necessary?

File: src/scss/_contact.scss [hunk 4]
```diff
@@ -1,87 +1,78 @@
...
-        }
+      &:focus {
+        outline: none;
       }
 
-      .message {
-        margin: 1.8rem 0;
-        resize: none;
+      &.disable:invalid {
+        border-color: $invalid;
```
Comment (rrudol) [inline src/scss/_contact.scss:49]: I think it is better to split this variable into plain params

File: src/scss/_contact.scss [hunk 5]
```diff
@@ -1,87 +1,78 @@
...
-          color: #fff;
-        }
-      }
+      background: $akai-yellow;
+      border: 0;
+      border-radius: 25px;
+      box-shadow: 0 5px 30px -5px $shadow;
+      color: $white-text;
+      cursor: pointer;
+      font-family: inherit;
```
Comment (rrudol) [inline src/scss/_contact.scss:67]: Isn't font-family global for all components?

File: src/scss/_contact.scss [hunk 6]
```diff
@@ -1,87 +1,78 @@
...
-        &:hover {
-          background: darken($akai-yellow, 6%);
-          color: #fff;
-        }
-      }
+      background: $akai-yellow;
+      border: 0;
+      border-radius: 25px;
+      box-shadow: 0 5px 30px -5px $shadow;
+      color: $white-text;
```
Comment (rrudol) [inline src/scss/_contact.scss:65]: Would $white-text be different than $white-background, I think we should use something like $akai-white

File: src/scss/_contact.scss [hunk 7]
```diff
@@ -1,87 +1,78 @@
...
+      cursor: pointer;
+      font-family: inherit;
+      font-weight: bold;
+      padding: 15px 45px;
+      transition: all 0.4s ease-out;
```
Comment (rrudol) [inline src/scss/_contact.scss]: Is transition for all properties necessary?

File: src/scss/_date-and-venue.scss [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    justify-content: space-around;
+    margin: 0 auto;
+    max-width: 640px;
+  }
+
+  .column {
+    align-content: space-between;
+    display: flex;
+    flex-direction: column;
+    font-weight: 500;
```
Comment (rrudol) [inline src/scss/_date-and-venue.scss:17]: Does font contain such weight, or we should use here 400 or 700?

File: src/scss/_date-and-venue.scss [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    .date-value {
+      @include font-size(5.2);
+
+      line-height: 0.9em;
+    }
+
+    .venue {
+      @include font-size(2);
+
+      font-weight: 600;
```
Comment (rrudol) [inline src/scss/_date-and-venue.scss:52]: Does font contain such weight, or we should use here 400 or 700?

File: src/scss/_date-and-venue.scss [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    display: none;
+  }
+
+  .calendar-add,
+  .calendar-type {
+    @include font-size(1.6);
+
+    color: $akai-yellow;
+    cursor: pointer;
+    font-weight: 600;
```
Comment (rrudol) [inline src/scss/_date-and-venue.scss:67]: Does font contain such weight, or we should use here 400 or 700?

File: src/scss/_date-and-venue.scss [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    padding: 10px;
+    text-decoration: none;
+    transition: all 0.2s ease-in-out;
```
Comment (rrudol) [inline src/scss/_date-and-venue.scss]: as above

File: src/scss/_date-and-venue.scss [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+  .calendar-type {
+    margin: 0 30px;
+
+    i {
+      margin-right: 5px;
+    }
+  }
+
+  @media all and (max-width: 480px) {
```
Comment (rrudol) [inline src/scss/_date-and-venue.scss]: Could we use respond-to mixin?

File: src/scss/_home.scss [hunk 13]
```diff
@@ -1,13 +1,12 @@
...
 
-[class=home] {
+.home {
   background-attachment: fixed;
   background-image: url('../img/background.png');
   background-position: center;
   background-repeat: no-repeat;
   background-size: cover;
-  color: $home-color;
+  color: $white-text;
```
Comment (rrudol) [inline src/scss/_home.scss:9]: $akai-white?

File: src/scss/_home.scss [hunk 14]
```diff
@@ -16,153 +15,114 @@ $home-color-yellow: rgb(250, 162, 27);
...
       background-color: $akai-yellow;
-      border-color: $akai-yellow;
-      border-radius: 2em;
-      border-style: solid;
-      border-width: thin;
-      color: $home-color;
-      font-size: 0.9em;
+      border: 0;
+      border-radius: 25px;
+      color: $white-text;
```
Comment (rrudol) [inline src/scss/_home.scss]: $akai-white?

File: src/scss/_home.scss [hunk 15]
```diff
@@ -16,153 +15,114 @@ $home-color-yellow: rgb(250, 162, 27);
...
     transform: translateX(-50%);
-    transition: all, 0.4s, ease-out;
+    transition: all 0.3s ease-out;
 
     .home-icon-container {
       align-items: center;
+      cursor: pointer;
       display: flex;
       flex-direction: column;
+      font-weight: bold;
```
Comment (rrudol) [inline src/scss/_home.scss]: change this value into 700

File: src/scss/_home.scss [hunk 16]
```diff
@@ -16,153 +15,114 @@ $home-color-yellow: rgb(250, 162, 27);
...
       color: $akai-yellow;
     }
   }
-}
-
-@media all and (max-width: 1100px), (max-height: 700px) { // width < 1100px || height < 700px
-  [class=home] {
-    font-size: 0.9rem;
 
+  @media all and (max-width: 480px) {
```
Comment (rrudol) [inline src/scss/_home.scss]: respond-to?

**Review by rrudol — COMMENTED** [review]
Some imporvements, but minor

File: src/scss/_home.scss (hunk 13 above)
Comment (tomaszgil) [reply src/scss/_home.scss:9]: White text is not quite white, it's #eee. That's why we have two variables.

File: src/scss/_contact.scss (hunk 6 above)
Comment (tomaszgil) [reply src/scss/_contact.scss:65]: Yes, these colors differ.

File: src/scss/_contact.scss (hunk 5 above)
Comment (tomaszgil) [reply src/scss/_contact.scss:67]: We need that to override default browser behaviour. We do have font applied for both body and html element and without this line the font on the button is plain sans-serif.

File: src/scss/_date-and-venue.scss (hunk 8 above)
Comment (tomaszgil) [reply src/scss/_date-and-venue.scss:17]: It's quite strange, it seems like there's 400, 600, 700, but when I apply 500 as a font-weight it works - the weight is somwhere between 400 and 600. I'll investigate that alongside solving issue #33.

**Pull request merged**